hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/impl/BlockStreamManagerImpl.java

```diff
@@ -519,10 +528,25 @@ private static boolean impliesPostUpgradeWorkPending(
     }
 
     private boolean shouldCloseBlock(final long roundNumber, final int roundsPerBlock) {
+        // We need the signer to be ready
         if (!blockHashSigner.isReady()) {
             return false;
         }
-        return roundNumber % roundsPerBlock == 0 || roundNumber == freezeRoundNumber;
+
+        // During freeze round, we should close the block regardless of other conditions
+        if (roundNumber == freezeRoundNumber) {
+            return true;
+        }
+
+        // If blockPeriod is 0, use roundsPerBlock
+        if (blockPeriod == 0) {
```

Comment:
```suggestion
        if (blockPeriod <= 0) {
```
Just in case we accidentally get a bad config value, but probably not a big deal

Reply:
Updated